spec/models/job_spec.rb

```diff
@@ -30,9 +30,15 @@
     end
 
     it "allows run for any ref in refs params" do
-      job = FactoryGirl.create :job, project: project, refs: "master, staging"
+      job = FactoryGirl.create :job, project: project, refs: "master, staging, /testing.*/, /^unstable$/, /unstable-v[0-9]{1,}.[0-9]{1,}.[0-9]{1,}/"
       job.run_for_ref?("master").should be_true
       job.run_for_ref?("staging").should be_true
+      job.run_for_ref?("staging-v0.1.0").should be_false
+      job.run_for_ref?("testing").should be_true
+      job.run_for_ref?("testing-v0.1.0").should be_true
+      job.run_for_ref?("unstable").should be_true
+      job.run_for_ref?("unstable-v0.1.0").should be_true
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.